### PR TITLE
Allow users to clear code cache

### PR DESCRIPTION
**Pull request by deltakosh**

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Review by sebavan — APPROVED** [review]

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/15706/merge

Test environment: 
https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15706/merge/index.html

To test a playground add it to the URL, for example:

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15706/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/15706/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/15706/merge
https://gui.babylonjs.com/?snapshot=refs/pull/15706/merge
https://nme.babylonjs.com/?snapshot=refs/pull/15706/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/15706/merge#BCU1XR#0

File: packages/dev/core/src/Materials/effect.ts [hunk 1]
```diff
@@ -1448,6 +1448,17 @@ export class Effect implements IDisposable {
         return this;
     }
 
+    /**
+     * Use this wisely: It will remove the cached code from this effect
+     * It is probably ok to call it if you are not using ShadowDepthWrapper or if everything is already up and running
```
Comment (Popov72) [inline packages/dev/core/src/Materials/effect.ts:1453]: I think we should also add a comment stating that if lost contexts are to be managed, this method should not be called as we will need the shader source code to recreate the effects on context restoration.

**Comment by Popov72** [comment]
I also forgot one comment: when will a user call the new method? I'm not sure it's easy for them to call it at the right time, maybe we should add a property so that this is handled automatically (I think we'd call the method after the effect has been deemed ready)?

**Comment by bjsplat** [comment]
WebGL2 visualization test reporter:

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15706/merge/testResults/webgl2playwright/index.html

**Comment by bjsplat** [comment]
Visualization tests for WebGPU (Experimental)

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15706/merge/testResults/webgpuplaywright/index.html

**Review by Popov72 — APPROVED** [review]

**Pull request merged**